### ISSUE TITLE
"fsharp" tests are using FSharp.Core from a package

### ISSUE THREAD

**Issue by dsyme** [Microsoft/visualfsharp#6546]
It appears most of the "single file tests" in  "fsharp" test suite is now running using FSHarp.Core from a package rather than the freshly built on.  In #6345 I added some tracing to show the generated project files, some output below.  No explicit FSharp.Core is referenced and the critical line is:

          -r:C:\Users\VssAdministrator\.nuget\packages\fsharp.core\4.6.2\lib\net45\FSharp.Core.dll

More complete log for one test:

```
 ------------------ D:\a\1\s\tests\fsharp\core\quotes ---------------
 cd D:\a\1\s\tests\fsharp\core\quotes
 ------ Overrides file: D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\Directory.Overrides.targets
 <Project>
   <Target Name="Build" DependsOnTargets="RunFSharpScript" />
   <Target Name="Rebuild" DependsOnTargets="RunFSharpScript" />
 </Project>
 ------ Targets file: D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\Directory.Build.targets
 <Project>
   <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.targets', 'D:\a\1\s\tests\fsharp'))" />
   <Import Project="$(MSBuildThisFileDirectory)Directory.Overrides.targets" Condition="'$(OutputType)'=='Script'" />
 </Project>
 ------ Props file: D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\Directory.Build.props
 <Project>
   <PropertyGroup>
     <Configuration Condition="'$(Configuration)' == ''">release</Configuration>
     <FSharpTestCompilerVersion>net40</FSharpTestCompilerVersion>
   </PropertyGroup>
   <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', 'D:\a\1\s\tests\fsharp'))" />
 </Project>
 ------ Project file: D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\e3v40p0b.szj.fsproj
 <Project Sdk='Microsoft.NET.Sdk'>
         
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net472</TargetFramework>
     <IsPackable>false</IsPackable>
     <DebugSymbols>True</DebugSymbols>
     <DebugType>portable</DebugType>
     <Optimize>True</Optimize>
     <SignAssembly>false</SignAssembly>
     <DefineConstants>FX_RESHAPED_REFLECTION</DefineConstants>
     <DefineConstants Condition="'$(OutputType)' == 'Script' and '$(FSharpTestCompilerVersion)' == 'coreclr'">NETCOREAPP</DefineConstants>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <RestoreAdditionalProjectSources Condition = " '$(RestoreAdditionalProjectSources)' == ''">D:\a\1\s\tests\fsharp/../../artifacts/packages/release</RestoreAdditionalProjectSources>
     <RestoreAdditionalProjectSources Condition = " '$(RestoreAdditionalProjectSources)' != ''">$(RestoreAdditionalProjectSources);D:\a\1\s\tests\fsharp/../../artifacts/packages/release</RestoreAdditionalProjectSources>
   </PropertyGroup>
 
   <!-- Utility sources -->
   <ItemGroup>
     <Compile Include='D:\a\1\s\tests\fsharp\coreclr_utilities.fs' />
   </ItemGroup>
 
   <!-- Sources -->
   <ItemGroup>
   </ItemGroup>
 
   <!-- Extra sources -->
   <ItemGroup>
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\testlib.fsi' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\testlib.fsi')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\testlib.fs' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\testlib.fs')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test.mli' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test.mli')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test.ml' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test.ml')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test.fsi' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test.fsi')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test.fs' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test.fs')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test2.fsi' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test2.fsi')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test2.fs' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test2.fs')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test.fsx' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test.fsx')" />
     <Compile Include='D:\a\1\s\tests\fsharp\core\quotes\test2.fsx' Condition="Exists('D:\a\1\s\tests\fsharp\core\quotes\test2.fsx')" />
   </ItemGroup>
 
   <!-- References -->
   <ItemGroup>
     <Reference Condition="$(TargetFramework.StartsWith('net4'))" Include="System.Windows.Forms" />
     <Reference Condition="$(TargetFramework.StartsWith('net4'))" Include="System.Web" />
   </ItemGroup>
 
   <Target Name='CopyCustomContentOnPublish' AfterTargets='Build'>
     <ItemGroup>
         <Libraries Include='*.dll' />
     </ItemGroup>
     <Copy SourceFiles='@(Libraries)' DestinationFolder='$(OutputPath)' SkipUnchangedFiles='false' />
   </Target>
 
 </Project>
 ------ Execute:
 dotnet.exe run -f net472 -v:n
 Build started 4/14/2019 9:39:15 PM.
      1>Project "D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\e3v40p0b.szj.fsproj" on node 1 (Restore target(s)).
      1>Restore:
          Restoring packages for D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\e3v40p0b.szj.fsproj...
          Committing restore...
          Generating MSBuild file D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\obj\e3v40p0b.szj.fsproj.nuget.g.props.
          Generating MSBuild file D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\obj\e3v40p0b.szj.fsproj.nuget.g.targets.
          Writing lock file to disk. Path: D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\obj\project.assets.json
          Restore completed in 183.35 ms for D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\e3v40p0b.szj.fsproj.
          
          NuGet Config files used:
              D:\a\1\s\NuGet.Config
              C:\Users\VssAdministrator\AppData\Roaming\NuGet\NuGet.Config
              C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config
              C:\Program Files (x86)\NuGet\Config\Xamarin.Offline.config
          
          Feeds used:
              https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json
              https://api.nuget.org/v3/index.json
              D:\a\1\s\artifacts\packages\release
      1>Done Building Project "D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\e3v40p0b.szj.fsproj" (Restore target(s)).
 
 Build succeeded.
     0 Warning(s)
     0 Error(s)
 
 Time Elapsed 00:00:00.89
 Build started 4/14/2019 9:39:16 PM.
      1>Project "D:\a\1\s\artifacts\tmp\Release\rjzsta3c.vuj\e3v40p0b.szj.fsproj" on node 1 (default targets).
      1>GenerateFSharpTextResources:
          Creating directory "obj\release\net472\".
        PrepareForBuild:
          Creating directory "bin\release\net472\".
        GenerateBindingRedirects:
          No suggested binding redirects from ResolveAssemblyReferences.
        GenerateTargetFrameworkMonikerAttribute:
        Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
        CoreCompile:
          D:\a\1\s\artifacts\bin\fsc\release\net472\fsc.exe -o:obj\release\net472\e3v40p0b.szj.exe
          -g
          --debug:portable
          --noframework
          --define:FX_RESHAPED_REFLECTION
          --define:RELEASE
          --define:NETFRAMEWORK
          --define:NET472
          --optimize+
          --platform:anycpu
          -r:C:\Users\VssAdministrator\.nuget\packages\fsharp.core\4.6.2\lib\net45\FSharp.Core.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\mscorlib.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Core.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Data.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Drawing.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.IO.Compression.FileSystem.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Numerics.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Runtime.Serialization.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Web.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Windows.Forms.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Xml.dll
          -r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Xml.Linq.dll
          --target:exe
          --warn:3
          --warnaserror:3239,76
          --fullpaths
          --flaterrors
          --subsystemversion:6.00
          --highentropyva+
          --targetprofile:mscorlib
          --nocopyfsharpcore
          --simpleresolution
          D:\a\1\s\artifacts\tmp\Release\.NETFramework,Version=v4.7.2.AssemblyAttributes.fs
          D:\a\1\s\tests\fsharp\coreclr_utilities.fs
          D:\a\1\s\tests\fsharp\core\quotes\test.fsx
          
          
```

**Comment by dsyme** [comment]
(I will work on this)

**Comment by cartermp** [comment]
Fixed by #6550
